r2/kv/store.go

```diff
@@ -359,7 +356,7 @@ func (s *store) FetchRollupRuleHistory(namespaceID, rollupRuleID string) ([]*rul
 }
 
 func (s *store) newUpdateMeta(uOpts r2.UpdateOptions) rules.UpdateMetadata {
-	return s.updateHelper.NewUpdateMetadata(s.clockOpts.NowFn()().UnixNano(), uOpts.Author())
+	return s.updateHelper.NewUpdateMetadata(s.sOpts.ClockOptions().NowFn()().UnixNano(), uOpts.Author())
```

Comment:
nit: Conventionally we store the clock options as a cached field in the store so you don't need to make a function call every time.